### PR TITLE
443 add lumi calorimeters

### PR DESCRIPTION
**Pull request by dhevang**
### Briefly, what does this PR introduce?

Add Luminosity pair spectrometer calorimeter detector factories

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [ x ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### CODE REVIEW

File: src/detectors/LUMISPECCAL/CalorimeterHit_factory_EcalLumiSpecRecHits.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        m_pedMeanADC=100;//{this, "pedestalMean", 400};
+        m_pedSigmaADC=1;//{this, "pedestalSigma", 3.2};
+        m_resolutionTDC=10 * dd4hep::picosecond;//{this, "resolutionTDC", 10 * ps};
+
+        // zero suppression values
+        m_thresholdFactor=4.0;//{this, "thresholdFactor", 0.0};
+        m_thresholdValue=3.0;//{this, "thresholdValue", 0.0};
+
+        // energy correction with sampling fraction
+        m_sampFrac=0.998;//{this, "samplingFraction", 1.0};
```
Comment (veprbl) [inline src/detectors/LUMISPECCAL/CalorimeterHit_factory_EcalLumiSpecRecHits.h:40]: Value not known yet, I assume:
```suggestion
        m_sampFrac=1.0;
```

Comment (dhevang) [reply src/detectors/LUMISPECCAL/CalorimeterHit_factory_EcalLumiSpecRecHits.h:40]: Right, but I was just following the lead from other ECALs.  They seem (or seemed) to all have 0.998 as a default.

Comment (veprbl) [reply src/detectors/LUMISPECCAL/CalorimeterHit_factory_EcalLumiSpecRecHits.h:40]: I fully understand. We need to do this in a systematic way too. However, I'm looking to encourage detector experts to critically assess those values and then update/document them.

**Comment by veprbl** [comment]
```
for i in `cd src/detectors/EEMC; find -name '*.h'`; do echo %%%%%%%%%% ${i} %%%%%%%%%%; diff -u <(cat src/detectors/EEMC/${i} | sed -e 's/EcalEndcapN//' -e 's/EEMC//') <(cat src/detectors/LUMISPECCAL/${i/EcalEndcapN/EcalLumiSpec} | sed -e 's/EcalLumiSpec//' -e 's/LUMISPECCAL//'); echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%; done
```
```diff
%%%%%%%%%% ./Cluster_factory_EcalEndcapNClusters.h %%%%%%%%%%
--- /dev/fd/11	2023-05-03 16:27:57.690244874 -0400
+++ /dev/fd/12	2023-05-03 16:27:57.690624717 -0400
@@ -43,7 +43,7 @@
     void Init() override{
         auto app = GetApplication();
         //-------- Configuration Parameters ------------
-        m_input_simhit_tag="Hits";
+        m_input_simhit_tag="LumiSpecCALHits";
         m_input_protoclust_tag="IslandProtoClusters";
 
         m_sampFrac=1.0;//{this, "samplingFraction", 1.0};
@@ -93,7 +93,7 @@
 
         // Hand owner of algorithm objects over to JANA
         Set(m_outputClusters);
-        event->Insert(m_outputAssociations, "ClusterAssociations");
+        event->Insert(m_outputAssociations, "ClustersAssociations");
         m_outputClusters.clear(); // not really needed, but better to not leave dangling pointers around
         m_outputAssociations.clear();
     }
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%% ./CalorimeterHit_factory_EcalEndcapNRecHits.h %%%%%%%%%%
--- /dev/fd/11	2023-05-03 16:27:57.699192651 -0400
+++ /dev/fd/12	2023-05-03 16:27:57.699652631 -0400
@@ -37,18 +37,17 @@
         m_thresholdValue=3.0;//{this, "thresholdValue", 0.0};
 
         // energy correction with sampling fraction
-        m_sampFrac=0.998;//{this, "samplingFraction", 1.0};
+        m_sampFrac=1.0;
 
         // geometry service to get ids, ignored if no names provided
         m_geoSvcName="geoServiceName";
-        m_readout="Hits";  // from ATHENA's reconstruction.py
+        m_readout="LumiSpecCALHits";  // from ATHENA's reconstruction.py
         m_layerField="";              // from ATHENA's reconstruction.py (i.e. not defined there)
         m_sectorField="sector";       // from ATHENA's reconstruction.py
 
         m_localDetElement="";         // from ATHENA's reconstruction.py (i.e. not defined there)
         u_localDetFields={};          // from ATHENA's reconstruction.py (i.e. not defined there)
 
-//        app->SetDefaultParameter(":tag",              m_input_tag);
         app->SetDefaultParameter(":RecHits:input_tag",        m_input_tag, "Name of input collection to use");
         app->SetDefaultParameter(":RecHits:capacityADC",      m_capADC);
         app->SetDefaultParameter(":RecHits:dynamicRangeADC",  m_dyRangeADC);
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%% ./Cluster_factory_EcalEndcapNTruthClusters.h %%%%%%%%%%
--- /dev/fd/11	2023-05-03 16:27:57.708509713 -0400
+++ /dev/fd/12	2023-05-03 16:27:57.708836232 -0400
@@ -42,7 +42,7 @@
     void Init() override{
         auto app = GetApplication();
         //-------- Configuration Parameters ------------
-        m_input_simhit_tag="Hits";
+        m_input_simhit_tag="LumiSpecCALHits";
         m_input_protoclust_tag="TruthProtoClusters";
 
         m_sampFrac=1.0;//{this, "samplingFraction", 1.0};
@@ -92,7 +92,7 @@
 
         // Hand owner of algorithm objects over to JANA
         Set(m_outputClusters);
-        event->Insert(m_outputAssociations, "TruthClusterAssociations");
+        event->Insert(m_outputAssociations, "TruthClustersAssociations");
         m_outputClusters.clear(); // not really needed, but better to not leave dangling pointers around
         m_outputAssociations.clear();
     }
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%% ./ProtoCluster_factory_EcalEndcapNIslandProtoClusters.h %%%%%%%%%%
--- /dev/fd/11	2023-05-03 16:27:57.716280287 -0400
+++ /dev/fd/12	2023-05-03 16:27:57.716633112 -0400
@@ -34,14 +34,14 @@
 
         // adjacency matrix
         m_geoSvcName = "GeoSvc";
-        u_adjacencyMatrix = "(abs(row_1 - row_2) + abs(column_1 - column_2)) == 1";
+        u_adjacencyMatrix = "(sector_1 == sector_2) && ((abs(ceil(module_1 / 10) - ceil(module_2 / 10)) + abs(fmod(module_1, 10) - fmod(module_2, 10))) == 1)";
         u_adjacencyMatrix.erase(
           std::remove_if(u_adjacencyMatrix.begin(), u_adjacencyMatrix.end(), ::isspace),
           u_adjacencyMatrix.end());
-        m_readout = "Hits";
+        m_readout = "LumiSpecCALHits";
 
         // neighbour checking distances
-        m_sectorDist=5.0 * dd4hep::cm;             // from ATHENA reconstruction.py
+        m_sectorDist=0.0 * dd4hep::cm;             // from ATHENA reconstruction.py
         u_localDistXY={};     //{this, "localDistXY", {}};
         u_localDistXZ={};     //{this, "localDistXZ", {}};
         u_localDistYZ={};     //{this, "localDistYZ", {}};
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%% ./RawCalorimeterHit_factory_EcalEndcapNRawHits.h %%%%%%%%%%
--- /dev/fd/11	2023-05-03 16:27:57.724241594 -0400
+++ /dev/fd/12	2023-05-03 16:27:57.724598094 -0400
@@ -35,7 +35,7 @@
         auto app = GetApplication();
 
         // Set default values for all config. parameters in CalorimeterHitDigi algorithm
-        m_input_tag = "Hits";
+        m_input_tag = "LumiSpecCALHits";
         u_eRes = {0.0 * sqrt(dd4hep::GeV), 0.02, 0.0 * dd4hep::GeV}; // flat 2%
         m_tRes = 0.0 * dd4hep::ns;
         m_capADC = 16384;
@@ -53,7 +53,6 @@
 
 
         // This is another option for exposing the data members as JANA configuration parameters.
-//        app->SetDefaultParameter(":tag",              m_input_tag);
         app->SetDefaultParameter(":RawHits:input_tag",        m_input_tag, "Name of input collection to use");
         app->SetDefaultParameter(":RawHits:energyResolutions",u_eRes);
         app->SetDefaultParameter(":RawHits:timeResolution",   m_tRes);
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%% ./ProtoCluster_factory_EcalEndcapNTruthProtoClusters.h %%%%%%%%%%
--- /dev/fd/11	2023-05-03 16:27:57.731466795 -0400
+++ /dev/fd/12	2023-05-03 16:27:57.731779214 -0400
@@ -27,7 +27,7 @@
     void Init() override{
         auto app = GetApplication();
         m_inputHit_tag="RecHits";
-        m_inputMCHit_tag="Hits";
+        m_inputMCHit_tag="LumiSpecCALHits";
 
         app->SetDefaultParameter(":TruthProtoClusters:inputHit_tag",        m_inputHit_tag, "Name of input collection to use");
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Review by veprbl — APPROVED** [review]

**Review by veprbl — APPROVED** [review]

**Pull request merged**